presto-main/src/main/java/com/facebook/presto/cost/HistoryBasedPlanStatisticsCalculator.java

```diff
@@ -201,7 +202,7 @@ private PlanNodeStatsEstimate getStatistics(PlanNode planNode, Session session,
                         Optional<HistoricalPlanStatisticsEntry> historicalPlanStatisticsEntry = getSelectedHistoricalPlanStatisticsEntry(entry.getValue(), inputTableStatistics.get(), historyMatchingThreshold);
                         if (historicalPlanStatisticsEntry.isPresent()) {
                             PlanStatistics predictedPlanStatistics = historicalPlanStatisticsEntry.get().getPlanStatistics();
-                            if (predictedPlanStatistics.getConfidence() > 0) {
+                            if (toConfidenceLevel(predictedPlanStatistics.getConfidence()).ordinal() >= delegateStats.confidenceLevel().ordinal()) {
```

Comment:
Why not add a comparator to the confidence level class? It will be useful in future comparisions